src/main/java/org/elasticsearch/discovery/srv/SrvUnicastHostsProvider.java

```diff
@@ -122,6 +122,13 @@ protected Resolver buildResolver(Settings settings) {
                 logger.warn("Could not create resolver. Using default resolver.", e);
             }
         }
+
+        parent_resolver = parent_resolver == null ? Lookup.getDefaultResolver() : parent_resolver;
```

Comment:
That is an unsettling thought indeed, damn globals. I think if/when that becomes a reality, the solution will be to implement the ability to specify a protocol per server, and tell the user to specify their servers explicitly.